frontend/packages/console-dynamic-plugin-sdk/src/utils/fetch/console-fetch.ts

```diff
@@ -54,13 +61,20 @@ const consoleFetchCommon = async (
  * @param method  The HTTP method to use. Defaults to GET
  * @param options The options to pass to fetch
  * @param timeout The timeout in milliseconds
- * @returns A promise that resolves to the response as JSON object.
+ * @param isEntireResponse The flag to control whether to return the entire content of the response or response body. The default is the response body.
+ * @returns A promise that resolves to the response as text, response JSON object or entire content of the HTTP response.
```

Comment:
Is this accurate? Shouldn't we only get either JSON or the entire response?

Reply:
> Is this accurate? Shouldn't we only get either JSON or the entire response?

Your earlier comment mentioned text, JSON or the entire response. I think this is in order even though JSON type is not explicitly in the type return of `consoleFetchCommon`. Anyone reading the code would know that the `response` is parsed as JSON.